code-challenges.js

```diff
@@ -8,62 +8,160 @@
+
+const boilingPoint = temp => {
+    if (temp < 212){
+        return `${temperature1} is below boiling point`;
+    }
+    else if (temp > 212){
+        return `${temperature2} is above boiling point`;
+    }
+    else  {
+        return `${temperature3} is at boiling point`;
+    }
+}
+
 
 const temperature1 = 42
 // Expected output: "42 is below boiling point"
+console.log(boilingPoint(temperature1))
+   // output: 42 is below boiling point
 
 const temperature2 = 350
 // Expected output: "350 is above boiling point"
-
+console.log(boilingPoint(temperature2))
+// output: 350 is above boiling point
 const temperature3 = 212
 // Expected output: "212 is at boiling point"
+console.log(boilingPoint(temperature3))
+// output: 212 is at boiling point
 
 // --------------------2) Create the code that will combine the two arrays and return the length using the test variables provided below.
 
 // Pseudo code:
+//-input is 2 arrays 
+//-output return length of both arrays combined using .length built in method
+//-process create a function combinedArrays and set parameters to combine arrays
+//-return the output of the combined arrays a + b
+const combinedArrays = (a,b) => {
+    return a + b
+}
 
 const padres1984WorldSeriesRuns = [2, 5, 2, 2, 4]
 const padres1998WorldSeriesRuns = [6, 3, 5, 3]
 // Expected output: 9
 
+console.log(combinedArrays(padres1984WorldSeriesRuns.length, padres1998WorldSeriesRuns.length))
+//output: 9
+
 // --------------------3) Create the code that will reverse the letters of a string using the test variable provided below.
 
 // Pseudo code:
+//-input is the test variable Charlie 2023
+//-output is the string reversed
+//-process create a function called stringReversed and set only 1 parameter to equal the number of test variables, then turn string into an array, reverse the array and join the array back into a string
+//-return the string reversed using the .split built in method to create the array, .reverse to reverse the array and then use the .join built in method to join the array into a string
+
+const stringReversed = (a) => {
+    return a.split("").reverse().join("")
+
+}
 
 const currentCohort = "Charlie 2023"
 // Expected output: "3202 eilrahC"
 
+
+console.log(stringReversed(currentCohort))
+  //output" 3202 eilrahC
+
+
 // --------------------4) Create the code that will return the last index of the given value from the array using the test variables provided below.
 
 // Pseudo code:
+// -input is the array numberOfConnections
+// -output is the last index of the givenValue1 and givenValue2
+// -process is to create a function named lastInstance, search numberOfConnections for the last index of givenValue1 and givenValue2 using the .lastIndexOf built in method give it a single parameter and use the function to index the given value
+// -return the expected output of givenValue1 and givenValue2 respectively
+
+const lastInstance = (a) => {
+    return numberOfConnections.lastIndexOf(a)
+}
```

Comment:
Same thing here, you don't want to use the actual test variables inside your function, since you want your function to be dynamic. Going along with what I said in the previous comment, the number of parameters is determined by how many inputs there will be for each function call. So in this case, your function should have 2 parameters, the array, and the given value. From there, you can make the function dynamic using those 2 parameters.